openmicroscopy/bioformats
# BF Convert - Optional Zero Padding

This PR is aimed to add a zero padding option when using bfconvert tool.
- When the -padded option is set the filename parameters for series, z, c and t should be padded with zeros. The padding length is not configurable but is based on the image parameters.
- When the new -padded option is not set the default behaviour should be preserved 

The request has been raised in various places:
https://github.com/openmicroscopy/bioformats/issues/2490
https://trac.openmicroscopy.org/ome/ticket/12678
http://lists.openmicroscopy.org.uk/pipermail/ome-users/2016-August/006130.html

To test:
- With the new parameter set verify that bfconvert (on an image with a dimension or series parameter > 10) will be padded with zeros:
For example - bfconvert  testing.nd2 output_xy%sz%zc%ct%t.ome.tif -padded
Should result in filenames such as:

| Original | New Padded name |
|---------|-----------------------|
| output_xy0z0c0t0.ome.tif | output_xy0z000c0t0.ome.tif |
| output_xy0z9c0t0.ome.tif | output_xy0z009c0t0.ome.tif |
| output_xy0z10c0t0.ome.tif | output_xy0z010c0t0.ome.tif |
| output_xy0z19c0t0.ome.tif | output_xy0z019c0t0.ome.tif |
| output_xy0z20c0t0.ome.tif | output_xy0z020c0t0.ome.tif |
| output_xy0z100c0t0.ome.tif | output_xy0z100c0t0.ome.tif |

- Verify that the original default behaviour still works as before when the new option is not set
